frontend/src/components/CardDetails/CardDetails.test.tsx

```diff
@@ -1,55 +1,54 @@
-import { render, screen } from "@testing-library/react";
-import { MemoryRouter } from "react-router";
-import { describe, expect, it } from "vitest";
+import { screen } from "@testing-library/react";
 import { axe } from "vitest-axe";
 
+import { renderWithRouter } from "@/lib/test/renderWithRouter";
+
 import { CardDetails } from "./CardDetails";
 
-describe("Given a user is looking at an individual product's details", () => {
+describe("Given the user is looking at an individual products's details", () => {
   describe("When the product is displayed on the home page", () => {
-    it("Then store, product name and price is displayed", () => {
-      render(
-        <MemoryRouter initialEntries={["/"]}>
-          <CardDetails
-            productTitle="Pams Butter"
-            price="3.50"
-            store="Woolworths"
-          />
-        </MemoryRouter>,
+    it("Then the store, product name and price is displayed", () => {
+      renderWithRouter(
+        <CardDetails
+          productTitle="Pams Butter"
+          price="3.50"
+          store="Woolworths"
+        />,
```

Comment:
🐘 Could avoid repetition (and easier future updates for tests) also by doing: 

`const mockDetails = {
   productTitle="Pams Butter"
   price="3.50"
   store="Woolworths"
`

and then 

```
<CardDetails {...mockDetails} />
```